websites/S/Shikimori/presence.ts

```diff
@@ -3,152 +3,208 @@ const presence = new Presence({
-		document.location.pathname.includes("/ranobe/")
-	) {
-		presenceData.details = `Смотрит страницу ${document
-			.querySelector(".submenu-triangle > span")
-			.textContent.toLowerCase()}`;
-		presenceData.largeImageKey = (<HTMLImageElement>(
-			document.querySelector(".c-poster > center > img")
-		)).src;
-		presenceData.state = title;
-	} else if (
-		document.location.pathname.includes("/list/anime/mylist") ||
-		document.location.pathname.includes("/list/manga/mylist")
-	) {
-		if (document.location.pathname.includes("/planned"))
-			presenceData.state = "Запланированно";
-		else if (document.location.pathname.includes("/watching"))
-			presenceData.state = "Смотрю";
-		else if (document.location.pathname.includes("/rewatching"))
-			presenceData.state = "Пересматриваю";
-		else if (document.location.pathname.includes("/completed"))
-			presenceData.state = "Просмотрено";
-		else if (document.location.pathname.includes("/on_hold"))
-			presenceData.state = "Отложено";
-		else if (document.location.pathname.includes("/dropped"))
-			presenceData.state = "Брошено";
+	function isImageExist(tags: string) {
+		return document?.querySelector<HTMLImageElement>(tags) && !privacy && logo
+			? document?.querySelector<HTMLImageElement>(tags)?.src
+			: "shikimori_logo";
+	}
 
-		presenceData.details = `Смотрит ${document
-			.querySelector(".submenu-triangle > span")
-			.textContent.toLowerCase()} ${document.location.pathname.split("/")[1]}`;
-		presenceData.largeImageKey = (<HTMLImageElement>(
-			document.querySelector(".block.avatar center a img")
-		)).src;
-	} else if (document.location.pathname === `/${title}`) {
-		presenceData.details = `В профиле ${title}`;
-		presenceData.largeImageKey = (<HTMLImageElement>(
-			document.querySelector(".avatar img")
-		)).src;
-	} else if (
-		document.location.pathname.includes("/list/anime/") ||
-		document.location.pathname.includes("/list/manga/")
-	) {
-		presenceData.details = `Смотрит ${document
-			.querySelector(".submenu-triangle > span")
-			.textContent.toLowerCase()} ${document.location.pathname.split("/")[1]}`;
-		presenceData.largeImageKey = (<HTMLImageElement>(
-			document.querySelector(".block.avatar center a img")
-		)).src;
-	} else if (
-		document.location.pathname.includes("/characters/") ||
-		document.location.pathname.includes("/people/")
-	) {
-		let type;
-		if (document.location.pathname.includes("/characters/")) type = "персонажа";
-		else if (document.location.pathname.includes("/people/")) type = "человека";
-		presenceData.largeImageKey = (<HTMLImageElement>(
-			document.querySelector(".c-poster center img")
-		)).src;
-		presenceData.details = `Смотрит страницу ${type}`;
-		presenceData.state = title;
+	if (buttons && !privacy) {
+		presenceData.buttons = [
+			{
+				label: "Открыть страницу",
+				url: document
+					.querySelector("meta[property='og:url']")
+					.getAttribute("content"),
+			},
+		];
+	}
+
+	switch (document.location.pathname.split("/")[1]) {
```

Comment:
`document.location.pathname` is used a lot. Consider destructuring it instead.

Reply:
> `document.location.pathname` is used a lot. Consider destructuring it instead.

Okay, I'll change it.